pennylane/qnodes/passthru.py

```diff
@@ -115,7 +115,11 @@ def _construct(self, args, kwargs):
                 pennylane.operation.Operator.do_check_domain = True
 
         # check the validity of the circuit
+        # turn off domain checking, but outside of the context such that no
+        # queuing takes place (e.g. from decompositions)
+        pennylane.operation.Operator.do_check_domain = False
         self._check_circuit(res)
+        pennylane.operation.Operator.do_check_domain = True
```

Comment:
It might be a good idea to place this in a `try`/`finally` block. That way, if `self._check_circuit()` errors, and the user manually catches the exception, the class attribute `pennylane.operation.Operator.do_check_domain` is properly reset to `True`:

```python
try:
    pennylane.operation.Operator.do_check_domain = False
    self._check_circuit(res)
finally:
    pennylane.operation.Operator.do_check_domain = True
```

Reply:
Note: this feels like a temporary fix, and in the long run we should revisit the operation parameter checks.